backend/src/routes/api/storage/index.ts

```diff
@@ -0,0 +1,105 @@
+import { FastifyInstance, FastifyReply } from 'fastify';
+import { getObjectSize, getObjectStream, setupMinioClient } from './storageUtils';
+import { getDashboardConfig } from '../../../utils/resourceUtils';
+import { getDirectCallOptions } from '../../../utils/directCallUtils';
+import { getAccessToken } from '../../../utils/directCallUtils';
+import { OauthFastifyRequest } from '../../../types';
+
+export default async (fastify: FastifyInstance): Promise<void> => {
+  fastify.get(
+    '/:namespace/size',
+    async (
+      request: OauthFastifyRequest<{
+        Querystring: { key: string };
+        Params: { namespace: string };
+      }>,
+      reply: FastifyReply,
+    ) => {
+      try {
+        const dashConfig = getDashboardConfig();
+        if (dashConfig?.spec.dashboardConfig.disableS3Endpoint !== false) {
+          reply.code(404).send('Not found');
+          return reply;
+        }
+
+        const { namespace } = request.params;
+        const { key } = request.query;
+
+        const requestOptions = await getDirectCallOptions(fastify, request, request.url);
+        const token = getAccessToken(requestOptions);
+
+        const { client, bucket } = await setupMinioClient(fastify, token, namespace);
+
+        const size = await getObjectSize({
+          client,
+          key,
+          bucket,
+        });
+
+        reply.send(size);
+      } catch (err) {
+        reply.code(500).send(err.message);
+        return reply;
+      }
+    },
+  );
+
+  fastify.get(
+    '/:namespace',
+    async (
+      request: OauthFastifyRequest<{
+        Querystring: { key: string; peek?: number };
+        Params: { namespace: string };
+      }>,
+      reply: FastifyReply,
+    ) => {
+      try {
+        const dashConfig = getDashboardConfig();
+        if (dashConfig?.spec.dashboardConfig.disableS3Endpoint !== false) {
+          reply.code(404).send('Not found');
+          return reply;
+        }
+
+        const { namespace } = request.params;
+        const { key, peek } = request.query;
```

Comment:
We need to make sure peek doesn't exceed our limit -- it doesn't matter what the UI says